### ISSUE TITLE
Support multiple scope names in a single keyscope attribute

### ISSUE THREAD

**Issue by raducoravu** [dita-ot/dita-ot#2004]
I publish the following DITA Map which has key scopes and references in it to XHTML:

http://www.oxygenxml.com/forum/files/dot-bugs-testKSNPE.zip

and I obtain at some point:

            BUILD FAILED
            D:\projects\eXml\frameworks\dita\DITA-OT2.x\build.xml:41: The following error occurred while executing this line:
            D:\projects\eXml\frameworks\dita\DITA-OT2.x\plugins\org.dita.base\build_preprocess.xml:268: java.lang.NullPointerException
            	at org.dita.dost.module.KeyrefModule.processFile(KeyrefModule.java:209)
            	at org.dita.dost.module.KeyrefModule.execute(KeyrefModule.java:84)
            	at org.dita.dost.pipeline.PipelineFacade.execute(PipelineFacade.java:68)

**Comment by jelovirt** [comment]
Marking as a feature, since this is for supporting multiple scope names in a single `@keyscope` attribute.

**Comment by raducoravu** [comment]
Ah, that's the cause of the NPE then.

**Comment by jelovirt** [comment]
@raducoravu I fixed the implementation to match what unit tests would expect. Could you test the code now and see if it matches your expectations?

**Comment by raducoravu** [comment]
I tested publishing to XHTML and I think this properly works now. I can make keyrefs with both key scope names and they all seem to resolve properly. Later on I will probably try to create and test some more complicated examples, with multiple keyscopes which engulf each other, I will add another issue if I find anything.
I wonder why the attribute was not called @keyscopes instead of @keyscope.

**Comment by jelovirt** [comment]
Closing this as fixed.

WRT naming the attributes, there's already `@keys` that has the plural there. You're right, it would have made more sense to call it `@keyscopes`, but it's too late to change not. cc @robander

**Comment by robander** [comment]
Would need to ask Chris Nitchie about the name, but I think it's because originally there was an expectation that it would not have multiple values - so original design used `@keyscope` singular, and that never changed. I do think that using a single value will be more common, but using one key is more common and we still used `@keys

**Comment by ctnitchie** [comment]
@robander is entirely correct. In the original proposal there was no intra-scope key referencing, and so the value of `@keyscope` was simply an identifier for use in debugging/reporting. The presence of the `@keyscope` attribute was all that was required for the processing. Then we decided to add cross-scope references, then we realized that it really should allow multiple names to follow the precedent of `@keys`, but we never went back and updated the name.
